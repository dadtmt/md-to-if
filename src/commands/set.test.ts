import { getCommand, applyCommand } from '.'

describe('applyCommand set', () => {
  it('returns [empty text content node, state with property setted] for instruction set', () => {
    const currentSceneName = 'currentSceneName'

    const content = {
      content: [
        {
          content: ' set container prop val value',
          type: 'text',
        },
      ],
      type: 'command',
    }

    const state = {
      played: {
        currentSceneName: 3,
      },
      currentSceneName,
    }

    const expected = [
      {
        content: '',
        type: 'text',
      },
      {
        played: {
          currentSceneName: 3,
        },
        currentSceneName,
        container: { prop: 'value' },
      },
    ]

    expect(applyCommand(state)(content)).toEqual(expected)
  })
  it('returns [error node content with missing path message, unmodified state]', () => {
    const content = {
      content: [
        {
          content: ' set val 2 ',
          type: 'text',
        },
      ],
      type: 'command',
    }
    const state = { currentSceneName: 'current scene' }

    expect(applyCommand(state)(content)).toMatchSnapshot()
  })
})
