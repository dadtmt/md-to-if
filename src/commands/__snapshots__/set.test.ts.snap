// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyCommand set returns [error node content with missing path message, unmodified state] 1`] = `
Array [
  Object {
    "content": Array [
      Object {
        "content": "path is required to set a value -- path val value",
        "type": "text",
      },
      Object {
        "content": Array [
          Object {
            "content": " set val 2 ",
            "type": "text",
          },
        ],
        "type": "command",
      },
    ],
    "type": "error",
  },
  Object {
    "currentSceneName": "current scene",
  },
]
`;
